<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Protected Data Usage Guide</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-dark.css">
    <script src="javascripts/scale.fix.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">

    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>Protected Data Usage Guide</h1>
        <p class="view"><a href="https://github.com/ncbi/sra-tools">View the Project on GitHub <small>ncbi/sra-tools</small></a></p>
        <ul>
          <li><a href="https://github.com/ncbi/sra-tools/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/ncbi/sra-tools/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/ncbi/sra-tools">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
	  	<p>
	  		The following guide will outline the configuration of the SRA Toolkit for use with protected data from dbGaP. Detailed information regarding the usage of 				individual tools in the SRA Toolkit can be found on the tool-specific documentation pages.
		</p>
      </section>
	  <section>
	  	<h3>Steps for using the SRA Toolkit to access dbGaP data</h3>
	  		<ul>
	  			<li>User must have an up to date version of the <a href="http://www.ncbi.nlm.nih.gov/Traces/sra/?view=toolkit_doc">SRA Toolkit installed.</a></li>
	  			<li>
	  				Users that wish to access controlled-access data must first apply for approval. Please review the process at the 
	  				<a href="https://dbgap.ncbi.nlm.nih.gov/aa/wga.cgi?page=login">dbGaP Authorized Access page.</a>
	  			</li>
	  			<li>
	  				Once granted access to a project, the PI may login and click the "get dbGaP repository key" link next to the project to download the repository key. 					This file should be closely guarded.
	  			</li>
	  			<img src="images/dbgap_auth.png">
	  			<li>
	  				For users that do not yet have an approved project, the test key 
	  				<a href="ftp://ftp.ncbi.nlm.nih.gov/sra/examples/decrypt_examples/prj_phs710EA_test.ngc">prj_phs710EA_test.ngc</a>
	  				is available for accessing a copy of 1000 Genomes data from NCBI. Downloading this key will allow users to test their 
	  				toolkit configuration on encrypted data that is consented for public access.
	  			</li>
	  			<li>
	  				Import the dbGaP repository key to SRA Toolkit, either by <br>
	  				<strong>Visual SRA-Toolkit configuration</strong><br>
	  				<strong>Command line Toolkit configuration</strong>
	  			</li>	
	  		</ul>
	  	<h3>Visual SRA Toolkit configuration</h3>
	  		<ul>
	  			<li>
	  				Go to the "bin" subdirectory for the Toolkit and run the following command line: <br>
		  			<strong class="example">./vdb-config -i</strong>
	  			</li>
	  			<li>Review any settings presented to ensure they are correct. Then tab to "Import Repository Key" button and press enter or space.</li>
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/dbgap_1.png">
	  			<li>This will bring up the file navigation dialog (see below).</li>
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/dbgap_2.png">
	  			<li>Navigate to the location of the previously downloaded .ngc file for the project.</li>
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/dbgap_3.png">
	  			<li>and press tab to get to the file list and select the .ngc file.</li>
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/dbgap_4.png">
	  			<li>Press tab to get to "OK" button and press enter or space</li>
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/dbgap_5.png">
	  			<li>You will be prompted to confirm the import</li>
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/dbgap_6.png">
	  			<li>Then the import will be confirmed</li>
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/dbgap_7.png">
	  			<li>
	  				You will be asked to change the location where the projects files will be stored.<br>

	  				Genomics datasets are quite large; you may need 100's of GB of free space. This is the primary concern when choosing the location. 
	  				Do you have enough free space there for what you intend to do?<br>

	  				A workspace is the directory which contains all the data and analysis for a project. Each dbGaP project must have its own workspace 
	  				that is separate from other protected project workspaces.
	  			</li>
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/SRATree.png">
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/dbgap_8.png">
	  			<li>
	  				If you choose "Yes", this will bring up the file navigation dialog (see below). <br>

	  				If you already know the path to the directory, you may use the Goto button to directly enter that path.<br>

	  				Once you have entered or navigated to the correct directory, press tab to get to the OK button to return to the previous screen.
	  			</li>
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/dbgap_9.png">
	  			<li>Once you are back to the main screen, use tab to get to the "Exit" button and press enter or space</li>
	  			<img src="http://www.ncbi.nlm.nih.gov/Traces/sra/img/dbgap_10.png">
	  			<li>
	  				In order to access the projects data, you will need to change directory or "cd" to the project's workspace.<br>

	  				Once there, all the project's data is available to you. You do not need to (or even want to) decrypt it manually.	
	  			</li>
	  		</ul>
	  	<h3>Command line SRA Toolkit configuration</h3>
	  		<ul>
	  			<li>
	  				Use vdb-config to import the repository key. <br>

	  				<strong class="example">vdb-config --import prj_phs710EA_test.ngc</strong>	
	  			</li>
	  		</ul>
	  	<h3>Accessing Encrypted dbGaP Data</h3>
	  		<ul>
	  			<li>
	  				The SRA Toolkit <a href="http://www.ncbi.nlm.nih.gov/Traces/sra/?view=toolkit_doc&f=vdb-decrypt">vdb-decrypt</a>
	  				program supports the decryption process for dbGaP phenotype and genotype files, 
	  				however SRA files do not need to be decrypted, and the utility will ignore attempts to do so. <br>

	  				<strong class="example">vdb-decrypt < encrypted_file ></strong>
	  			</li>
	  			<li>
	  				VERY IMPORTANT - The SRA Toolkit will only decrypt and download project files when excecuted from within the project's workspace directory. 
	  				Below are some examples using the test key 
	  				<a href="ftp://ftp.ncbi.nlm.nih.gov/sra/examples/decrypt_examples/prj_phs710EA_test.ngc">prj_phs710EA_test.ngc</a> 
	  				running from the default import location for the repository key and the SRA toolkit available in the user's path. <br>
	  				Information about the <a href="http://www.ncbi.nlm.nih.gov/Traces/sra/?view=toolkit_doc&f=fastq-dump">fastq-dump</a> 
	  				program and options used in the example can be found on the tool's documentation page. <br>
	  				<strong class="example">~/ncbi/dbGaP-0 $ fastq-dump -Z -X 5 SRR1219902</strong>
	  			</li>
	  			<li>
	  				The <a href="http://www.ncbi.nlm.nih.gov/Traces/sra/?view=toolkit_doc&f=sam-dump">sam-dump</a> 
	  				utility can also be used to directly access aligned data in a dbGaP study without the need to download the data files first.<br>

	  				<strong class="example">~/ncbi/dbGaP-0 $ sam-dump --aligned-region 15:28196787-28197287 SRR1219902</strong>
	  			</li>
	  			<li>
	  				Some changes from earlier versions of the Toolkit:<br>
	  				<ul>
	  					<li>The SRA Toolkit will work with encrypted SRA data files, there is no need to decrypt the read data.</li>
	  					<li>Users do not need to explicitly download the SRR (SRA Run) files. The tools such as fastq-dump or sam-dump can download all or part of an SRR file and 					reference files automatically. Please see the usage instructions for the individual Toolkit utilities (e.g., fastq-dump) for details.</li>
	  				</ul>
	  				IMPORTANT: Do NOT run vdb-passwd from within the Download Repository directory (imported from .ngc). vdb-passwd is included in the package for 							backward compatibility.
	  			</li>
	  		</ul>
	  </section>
    </div>
    <footer>
      <p>Project maintained by <a href="https://github.com/ncbi">ncbi</a></p>
      <p>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></p>
    </footer>
    <!--[if !IE]><script>fixScale(document);</script><![endif]-->
    
  </body>
</html>
